src/mscorlib/corefx/System/Globalization/CalendarData.Unix.cs

```diff
@@ -193,8 +321,34 @@ private static bool EnumCalendarInfo(string localeName, CalendarId calendarId, C
 
         private static void EnumCalendarInfoCallback(string calendarString, IntPtr context)
         {
-            List<string> calendarDataList = (List<string>)((GCHandle)context).Target;
-            calendarDataList.Add(calendarString);
+            CallbackContext callbackContext = (CallbackContext)((GCHandle)context).Target;
+
+            if (callbackContext.DisallowDuplicates)
+            {
+                foreach (string existingResult in callbackContext.Results)
```

Comment:
can't use any C++ data structure like map/dictionary instead of looping in all stored items?